go/install/stop_linux.go

```diff
@@ -0,0 +1,53 @@
+package install
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+func StopAllButService(mctx libkb.MetaContext) {
+	clients := libkb.GetClientStatus(mctx)
+	for _, client := range clients {
+		if client.Details.ClientType == keybase1.ClientType_CLI {
+			continue
+		}
+
+		// NOTE KBFS catches the SIGTERM and attempts to unmount mountdir before terminating,
+		//      so we don't have to do it ourselves.
+		// NOTE We kill the GUI by its main electron process ID, which should
+		//      shut down its child processes.
+		err := stopPID(mctx, client.Details.Pid)
+		if err != nil {
+			mctx.Debug("Error killing client %+v: %s", client, err)
+		}
+	}
+
+	if mctx.G().Env.GetRunMode() == libkb.ProductionRunMode {
+		// NOTE killall only inspects the first 15 characters; we need to use pkill -f
+		err := stopProcessInexactMatch(mctx, "keybase-redirector")
```

Comment:
Doesn't `keybase-redirector` run as root? Will this actually work?

Reply:
no it's run by the user with suid